### PR TITLE
Address 542 duplicate phase naming and #551 slow unit tests

### PR DESCRIPTION
**Pull request by argerlt**
#### Description of the change
When loading a .ang file, ORIX will now either:

1) Do nothing if there is no formula data (this is the old behavior)
2) Rename Phases according to the `formula` value if all formula's are unique (ie "Fe", "Fe3C", .etc.)
3) If there are phases with identical formulas but different point groups, it will rename the phases as "formulat-PointGroup" (ie, `Ti-622` and `Ti-m-3m')

OR 

4) If you pass `'io.loadang('filename' ,autogen_names=False),  it overwrites nothing and uses whatever is already in the file.

@gsparks3, I think this PR addresses your problem of identically names phases. If you could try it out on a few files, that would be appreciated. it works correctly on [this data you posted in issue 543](https://github.com/user-attachments/files/19817756/TwoPhasesSameFormula_OIM8.6.ang.zip) as well as all the default ORIX datasets and unittests, but there might be more edge cases to consider.

@CSSFrancis  As an additional item to also close out #551, I reduced the run time of the 3 largest tests. Past that, it seemed to be a diminishing return and questionable equivalency. 

More importantly though, I updated the documentation to encourage developers to use pytest-xdist for parallel unit testing.  This is what the automated unit-tests have been using since 2022, so it seems safe to add to the development options. these two changes together took my unit testing time from 8 minutes to 40 seconds. 

#### Progress of the PR
- [x] [Docstrings for all functions](https://numpydoc.readthedocs.io/en/latest/example.html)
- [x] Unit tests with pytest for all lines
- [x] Clean code style by [running black via pre-commit](https://orix.readthedocs.io/en/latest/dev/code_style.html)

#### Minimal example of the bug fix or new feature
(assuming you have unzipped the .ang file discussed above)
```python
>>> from orix import io
>>> ebsd = io.load("TwoPhasesSameFormula_OIM8.6.ang")
>>> ebsd

Out[133]: 
Phase  Orientations    Name  Space group  Point group  Proper point group       Color
    1    63 (78.8%)  Ti-622         None          622                 622  tab:orange
    2    17 (21.2%)  Ti-432         None          432                 432    tab:blue
Properties: unknown1, unknown2, unknown3, unknown4, unknown5, unknown6, unknown7
Scan unit: um
```

#### For reviewers

- [ ] The PR title is short, concise, and will make sense 1 year later.
- [ ] New functions are imported in corresponding `__init__.py`.
- [ ] New features, API changes, and deprecations are mentioned in the unreleased
      section in `CHANGELOG.rst`.
- [ ] Contributor(s) are listed correctly in `__credits__` in `orix/__init__.py` and in
      `.zenodo.json`.

### CODE REVIEW

File: orix/io/plugins/ang.py [hunk 1]
```diff
@@ -355,10 +387,12 @@ def _get_phases_from_header(header: List[str]) -> dict:
...
 
-    # Use formulas in place of material names if they are all valid
+    # Pop out the formula information, which can optionally be used later to
+    # rename the phses
     formulas = phases.pop("formulas")
-    if len(formulas) == n_phases and all([len(name) for name in formulas]):
-        phases["names"] = formulas
+
+    # if len(formulas) == n_phases and all([len(name) for name in formulas]):
+    #     phases["names"] = formulas
```
Comment (CSSFrancis) [inline orix/io/plugins/ang.py:396]: This is left over and can be deleted.

**Review by CSSFrancis — APPROVED** [review]
@argerlt This looks great.  There is one bit of left over code that might be good to remove.

**Comment by argerlt** [comment]
@CSSFrancis , Added your suggestion and Co-pilot's.   

You could merge this as is, but probably makes sense to do #554 first, then I can change the wording in this test to appropriately match.

My fault for trying to tackle two tasks in one PR.
